pkg/controller/user-cluster-controller-manager/resources/resources/csi-migration/secret.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csimigration
+
+import (
+	"fmt"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/certificates/triple"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	certutil "k8s.io/client-go/util/cert"
+)
+
+var webhookConfig = fmt.Sprintf(`
+[WebHookConfig]
+port = "%d"
+cert-file = "/etc/webhook/cert.pem"
+key-file = "/etc/webhook/key.pem"
+`, resources.CSIMigrationWebhookPort)
+
+func TLSServingCertificateCreator(ca *triple.KeyPair) reconciling.NamedSecretCreatorGetter {
```

Comment:
Why are half the resources managed via an addon and some are managed via controllers?

Reply:
Since the CSI-related stuff is deployed as addons, I thought it was the best place to put this webhook too. The problem came when I needed to create the certificate signed by the CA and put it in the secret (and the CA bundle in the `webhookConfiguration`). I tried to do this in the addons controller, but I didn't find the way without re-writing a lot of code in the controller itself (we do not manage any certificate there). Then I moved the certificate handling in the user-cluster-controller-manager, where we do plenty of stuff similar to this.